### PR TITLE
Add support for resp3 protocol

### PR DESCRIPTION
**Pull request by monkey92t**
1. Support redis RESP3 protocol, can work under RESP2 and RESP3.
2. If redis-server version >=6.0, RESP3 protocol is enabled by default. Otherwise use RESP2 protocol (redis < 6.0).
3. Remove test `Subscribe __redis__:invalidate`, it only works in RESP2 mode(redis6.0+).
4. `$EOF:<40 bytes marker><CR><LF>` and `*?<CR><LF>` that have not been enabled by redis-server have not been implemented for the time being, they will be followed up in the future.

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -287,7 +289,7 @@ type Cmdable interface {
 	ClientList(ctx context.Context) *StringCmd
 	ClientPause(ctx context.Context, dur time.Duration) *BoolCmd
 	ClientID(ctx context.Context) *IntCmd
-	ConfigGet(ctx context.Context, parameter string) *SliceCmd
+	ConfigGet(ctx context.Context, parameter string) *StringStringMapCmd
```
Comment (vmihailenco) [inline commands.go:292]: MapStringStringCmd since we already have `MapStringInterfaceCmd`

**Comment by vmihailenco** [comment]
>If redis-server version >=6.0, RESP3 protocol is enabled by default. Otherwise use RESP2 protocol (redis < 6.0).

It looks like the code always uses https://redis.io/commands/hello to switch to RESP3. And `HELLO` is only available since v6.0.0. So I guess v9 branch
- does not support Redis < 6.0
- always uses RESP3

Which is fine - I just want to confirm my understanding.

**Comment by monkey92t** [comment]
> > If redis-server version >=6.0, RESP3 protocol is enabled by default. Otherwise use RESP2 protocol (redis < 6.0).
> 
> It looks like the code always uses https://redis.io/commands/hello to switch to RESP3. And `HELLO` is only available since v6.0.0. So I guess v9 branch
> 
> * does not support Redis < 6.0
> * always uses RESP3
> 
> Which is fine - I just want to confirm my understanding.

If conn.HELLO(ctx, 3) fails to execute (redis<6.0), there will be no error, and it will continue to use RESP2 to execute commands.
If redis-server >= 6.0, HELLO is executed successfully and RESP3 is used.

**Review by vmihailenco — APPROVED** [review]

**Comment by vmihailenco** [comment]
Awesome, thanks.

**Pull request merged**